Sming/Components/libsodium/.patches/libsodium.patch

```diff
@@ -0,0 +1,1232 @@
+-#define implementations_H
+-
+-int _crypto_generichash_blake2b_pick_best_implementation(void);
+-int _crypto_onetimeauth_poly1305_pick_best_implementation(void);
+-int _crypto_pwhash_argon2_pick_best_implementation(void);
+-int _crypto_scalarmult_curve25519_pick_best_implementation(void);
+-int _crypto_stream_chacha20_pick_best_implementation(void);
+-int _crypto_stream_salsa20_pick_best_implementation(void);
+-
+-#endif
+diff --git a/src/libsodium/include/sodium/runtime.h b/src/libsodium/include/sodium/runtime.h
+index 7f15d58e..863369f1 100644
+--- a/src/libsodium/include/sodium/runtime.h
++++ b/src/libsodium/include/sodium/runtime.h
+@@ -41,10 +41,6 @@ int sodium_runtime_has_aesni(void);
+ SODIUM_EXPORT_WEAK
+ int sodium_runtime_has_rdrand(void);
+ 
+-/* ------------------------------------------------------------------------- */
+-
+-int _sodium_runtime_get_cpu_features(void);
+-
+ #ifdef __cplusplus
+ }
+ #endif
+diff --git a/src/libsodium/include/sodium/utils.h b/src/libsodium/include/sodium/utils.h
+index ac801512..ba1f1a65 100644
+--- a/src/libsodium/include/sodium/utils.h
++++ b/src/libsodium/include/sodium/utils.h
+@@ -170,8 +170,6 @@ int sodium_unpad(size_t *unpadded_buflen_p, const unsigned char *buf,
+ 
+ /* -------- */
+ 
+-int _sodium_alloc_init(void);
+-
+ #ifdef __cplusplus
+ }
+ #endif
+diff --git a/src/libsodium/include/sodium/version.h b/src/libsodium/include/sodium/version.h
+new file mode 100644
+index 00000000..d0ee3f83
+--- /dev/null
++++ b/src/libsodium/include/sodium/version.h
+@@ -0,0 +1,32 @@
++#ifndef sodium_version_H
++#define sodium_version_H
++
++#include "export.h"
++
++#define SODIUM_VERSION_STRING "1.0.18"
++
++#define SODIUM_LIBRARY_VERSION_MAJOR 10
++#define SODIUM_LIBRARY_VERSION_MINOR 3
++
++
++#ifdef __cplusplus
++extern "C" {
++#endif
++
++SODIUM_EXPORT
++const char *sodium_version_string(void);
++
++SODIUM_EXPORT
++int         sodium_library_version_major(void);
++
++SODIUM_EXPORT
++int         sodium_library_version_minor(void);
++
++SODIUM_EXPORT
++int         sodium_library_minimal(void);
++
++#ifdef __cplusplus
++}
++#endif
++
++#endif
+diff --git a/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c b/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c
+index 6657e8e6..fc364da6 100644
+--- a/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c
++++ b/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c
+@@ -44,6 +44,10 @@
+ # include <poll.h>
+ #endif
+ 
++#ifdef ARCH_ESP8266
++#include <espinc/peri.h>
++#endif
++
+ #include "core.h"
+ #include "private/common.h"
+ #include "randombytes.h"
+@@ -108,7 +112,35 @@ randombytes_sysrandom_close(void)
+     return 0;
+ }
+ 
+-#else /* HAVE_SAFE_ARC4RANDOM */
++#elif defined(ARCH_ESP8266)
++
++static uint32_t
++randombytes_sysrandom(void)
++{
++    return RANDOM_REG32;
++}
++
++static void
++randombytes_sysrandom_stir(void)
++{
++}
++
++static void
++randombytes_sysrandom_buf(void * const buf, const size_t size)
```

Comment:
I think os_get_random() would be a better choice. It's also available in the emulator (and ESP32).